src/ChatManager.cpp

```diff
@@ -699,6 +699,37 @@ void whoisCommand(std::string full, std::vector<std::string>& args, CNSocket* so
     ChatManager::sendServerMessage(sock, "[WHOIS] Instance: " + std::to_string(PLAYERID(npc->instanceID)));
 }
 
+void lairUnlock(std::string full, std::vector<std::string>& args, CNSocket* sock) {
+    Player* plr = PlayerManager::getPlayer(sock);
+    BaseNPC* npc = NPCManager::getNearestNPC(plr->viewableChunks, plr->x, plr->y, plr->z);
```

Comment:
`getNearestNPC` doesn't ignore invisible NPCs like waypoints or terminators, so it's not going to get the correct NPC every time on its own.